TypeMap.empty.get[Any] throws fatal

```
import kyo.*

TypeMap.empty.get[Any]
```
```
Caused by: java.lang.RuntimeException: fatal: kyo.TypeMap of contents [TypeMap()] missing value of type: [scala.Any].
```

Either we need to make `empty` `Map(Tag[Any] -> ())` OR we need to special case.

--- a/kyo-data/shared/src/main/scala/kyo/TypeMap.scala
+++ b/kyo-data/shared/src/main/scala/kyo/TypeMap.scala
@@ -21,7 +21,8 @@ object TypeMap:
                 end while
                 fatal
             end search
-            self.getOrElse(t.erased, search).asInstanceOf[B]
+            if isEmpty && t =:= Tag[Any] then ().asInstanceOf[B]
+            else self.getOrElse(t.erased, search).asInstanceOf[B]
         end get
 
         inline def add[B](b: B)(using inline t: Tag[B]): TypeMap[A & B] =

--- a/kyo-data/shared/src/test/scala/kyo/TypeMapTest.scala
+++ b/kyo-data/shared/src/test/scala/kyo/TypeMapTest.scala
@@ -13,6 +13,9 @@ class TypeMapTest extends Test:
                   | TypeMap.empty.get[String]
                   |""".stripMargin)
         }
+        "get[Any]" in {
+            assert(TypeMap.empty.get[Any].isInstanceOf[Unit])
+        }
     }
     "single" - {
         "TypeMap[String]" in {

--- a/kyo-data/shared/src/test/scala/kyo/TypeMapTest.scala
+++ b/kyo-data/shared/src/test/scala/kyo/TypeMapTest.scala
@@ -9,9 +9,12 @@ class TypeMapTest extends Test:
             assert(TypeMap.empty.isEmpty)
         }
         "get[A]" in {
-            assertDoesNotCompile("""
-                  | TypeMap.empty.get[String]
-                  |""".stripMargin)
+            assertDoesNotCompile("TypeMap.empty.get[String]")
+            assertDoesNotCompile("TypeMap.empty.get[AnyVal]")
+            assertDoesNotCompile("TypeMap.empty.get[AnyRef]")
+            assertDoesNotCompile("TypeMap.empty.get[Matchable]")
+            assertDoesNotCompile("TypeMap.empty.get[Object]")
+            assertDoesNotCompile("TypeMap.empty.get[Nothing]")
         }
         "get[Any]" in {
             assert(TypeMap.empty.get[Any].isInstanceOf[Unit])

